docs/guide/advanced/navigation-failures.md

```diff
@@ -0,0 +1,57 @@
+# Navigation Failures
+
+When using `router-link`, Vue Router internally calls `router.push` to trigger a navigation. Depending on the current location and existing [Navigation Guards](./navigation-guards.md), this navigation might end up in a new page being shown, but there are a couple of situations where we will stay on the same page:
+
+- We are already on the page we are trying to go to
+- A navigation guard aborts the navigation by calling `next(false)`
+- A navigation guard throws an error or calls `next(new Error())`
+
+When using a regular `router-link`, **none of these failures will log an error**. However, if you are using `router.push` or `router.replace`, you might come across an _"Uncaught (in promise) Error"_ message followed by a more specific message in your console. This is part of the _Navigation Failures_ system in Vue Router and it is only available from version 3.2.0 onwards but existed for a long time before: through the two optional callbacks, `onComplete` and `onAbort` that can be passed to `router.push`. Since version 3.1.0, `router.push` and `router.replace` return a _Promise_ if no `onComplete`/`onAbort` callback is provided. This _Promise_ resolves instead of invoking `onComplete` and rejects instead of invoking `onAbort`.
```

Comment:
Would it make sense to split this into 2-3 paragraphs? 
1) About the error handling when using `router-link`.
2) Using the `onComplete` and `onAbort` callbacks.
"New in 3.1.0" tag
3) `router.push` and `router.replace` return a _Promise_.

Reply:
Since the Navigation failure system is only available from 3.4.0 (the next release), I don't think it makes much sense to add a new in 3.1.0 but instead we could put a New in 3.4.0 at the very top

How would you split into multiple paragraphs the explanation? They are all related sentences

Reply:
I think I should remove the whole reference to versions if I add something at the very top

Reply:
The thing is people often want to know why things are like that, that's why I added the version things were added at

Reply:
That makes sense. I wonder though if it is a good practice to mix historical context with informative docs. As in, the historical context might be very interesting to read but likely shouldn’t be required to understand what is happening. Maybe a ::tip:: with informations of how it changed across different versions?

Reply:
That's true, I updated that part, let me know what you think

Reply:
Looks great! Thank you!